### ISSUE TITLE
Handle GitHub OAuth restrictions gracefully

### ISSUE THREAD

**Issue by estib-vega** [gitbutlerapp/gitbutler#5190]
### Version

0.13.2

### Operating System

macOS

### Distribution Method

dmg (Apple Silicon)

### Discord message
https://discord.com/channels/1060193121130000425/1073202153163857920/1296480103785828386

It seems GitHub returns some API errors with warning, claiming that the user does have the correct credentials, but some restrictions apply.

Not super sure how we should handle this though.
Probably raise the message to the user and have the option to not display again.

### Error message
```
Although you appear to have the correct authorization credentials, the <ORG NAME> organization has enabled OAuth App access restrictions, meaning that data access to third-parties is limited.
```

**Comment by esolitos** [comment]
I assume this is referring to errors like the following, right?

> Although you appear to have the correct authorization credentials, the `octoorg` organization has enabled OAuth App access restrictions, meaning that data access to third-parties is limited.

**Comment by estib-vega** [comment]
> Although you appear to have the correct authorization credentials, the `octoorg` organization has enabled OAuth App access restrictions, meaning that data access to third-parties is limited.

@esolitos Correct. I'll add that message to the description to make this issue easier to identify :)

**Comment by esolitos** [comment]
Would it be an option to disable the check for PRs for a given project?

Having to close 5-6 large & permanent error popups each time I do a rebase is kinda annoying. 🥲

**Comment by shanujha** [comment]
Whenever I work on an organization's repository, I get an error message popup with every action I take. Even if I fork the repo to my personal account, the errors persist. Since this issue started, I haven’t been able to create pull requests to the organization’s repo at all—they fail with a similar error. Interestingly, everything works fine when I’m working on fully personal projects.

**Comment by noqcks** [comment]
Im getting this error 

> Although you appear to have the correct authorization credentials, the <ORG NAME> organization has enabled OAuth App access restrictions, meaning that data access to third-parties is limited.

Is there a way for my organization to add the gitbutler oauth app at all? I don't have admin permissions on my org and im not sure what to tell the GitHub admin.

**Comment by Byron** [comment]
The auth for the enterprise works with a personal access token, and no OAuth app is used. That way, no admin involvement is needed, but the app is basically you without distinction.

**Comment by estib-vega** [comment]
Also for non-Enterprise users, you are now able to authenticate through PAT, which *should* bypass the OAuth restrictions completely.
